src/wwwroot/js/genpage/main.js

```diff
@@ -692,30 +731,15 @@ function setCurrentImage(src, metadata = '', batchId = '', previewGrow = false,
     let extrasWrapper = isReuse ? document.getElementById('current-image-extras-wrapper') : createDiv('current-image-extras-wrapper', 'current-image-extras-wrapper');
     extrasWrapper.innerHTML = '';
     let buttons = createDiv(null, 'current-image-buttons');
-    let imagePathClean = src;
-    if (imagePathClean.startsWith("http://") || imagePathClean.startsWith("https://")) {
-        imagePathClean = imagePathClean.substring(imagePathClean.indexOf('/', imagePathClean.indexOf('/') + 2));
-    }
-    if (imagePathClean.startsWith('/')) {
-        imagePathClean = imagePathClean.substring(1);
-    }
-    if (imagePathClean.startsWith('Output/')) {
-        imagePathClean = imagePathClean.substring('Output/'.length);
-    }
-    if (imagePathClean.startsWith('View/')) {
-        imagePathClean = imagePathClean.substring('View/'.length);
-        let firstSlash = imagePathClean.indexOf('/');
-        if (firstSlash != -1) {
-            imagePathClean = imagePathClean.substring(firstSlash + 1);
-        }
-    }
+    let subButtons = [];
+    let imagePathClean = getImageFullSrc(src);
     let buttonsChoice = getUserSetting('ButtonsUnderMainImages', '');
     if (buttonsChoice == '')
     {
         buttonsChoice = defaultButtonChoices;
     }
     buttonsChoice = buttonsChoice.toLowerCase().replaceAll(' ', '').split(',');
-    let subButtons = [];
+    let isDataImage = src.startsWith('data:');
```

Comment:
(minor) as a matter of PR cleanliness, avoid unneeded edits to lines (ie put these back where they were, use the "Files" tab of your PR to see the actual edits as you're submitting them)